src/main/kotlin/org/openrewrite/RecipeOrigin.kt

```diff
@@ -10,6 +10,32 @@ class RecipeOrigin(
     val version: String,
     val jarLocation: URI
 ) {
+    private val artifactIdToBaseGithubUrl = mapOf(
+        "rewrite-circleci" to "https://github.com/openrewrite/rewrite-circleci/blob/main/src/main",
+        "rewrite-cloud-suitability-analyzer" to "https://github.com/openrewrite/rewrite-cloud-suitability-analyzer/blob/main/src/main",
+        "rewrite-concourse" to "https://github.com/openrewrite/rewrite-concourse/blob/main/src/main",
+        "rewrite-core" to "https://github.com/openrewrite/rewrite/blob/main/rewrite-core/src/main",
+        "rewrite-github-actions" to "https://github.com/openrewrite/rewrite-github-actions/blob/main/src/main",
+        "rewrite-gradle" to "https://github.com/openrewrite/rewrite/blob/main/rewrite-gradle/src/main",
+        "rewrite-groovy" to "https://github.com/openrewrite/rewrite/blob/main/rewrite-groovy/src/main",
+        "rewrite-hcl" to "https://github.com/openrewrite/rewrite/blob/main/rewrite-hcl/src/main",
+        "rewrite-java" to "https://github.com/openrewrite/rewrite/blob/main/rewrite-java/src/main",
+        "rewrite-java-security" to "https://github.com/openrewrite/rewrite-java-security/blob/main/src/main",
+        "rewrite-jhipster" to "https://github.com/openrewrite/rewrite-jhipster/blob/main/src/main",
+        "rewrite-json" to "https://github.com/openrewrite/rewrite/blob/main/rewrite-json/src/main",
+        "rewrite-kubernetes" to "https://github.com/openrewrite/rewrite-kubernetes/blob/main/src/main",
+        "rewrite-logging-frameworks" to "https://github.com/openrewrite/rewrite-logging-frameworks/blob/main/src/main",
+        "rewrite-maven" to "https://github.com/openrewrite/rewrite/blob/main/rewrite-maven/src/main",
+        "rewrite-micronaut" to "https://github.com/openrewrite/rewrite-micronaut/blob/main/src/main",
+        "rewrite-migrate-java" to "https://github.com/openrewrite/rewrite-migrate-java/blob/main/src/main",
+        "rewrite-properties" to "https://github.com/openrewrite/rewrite/blob/main/rewrite-properties/src/main",
+        "rewrite-quarkus" to "https://github.com/openrewrite/rewrite-quarkus/blob/main/src/main",
+        "rewrite-spring" to "https://github.com/openrewrite/rewrite-spring/blob/main/src/main",
+        "rewrite-terraform" to "https://github.com/openrewrite/rewrite-terraform/blob/main/src/main",
+        "rewrite-testing-frameworks" to "https://github.com/openrewrite/rewrite-testing-frameworks/blob/main/src/main",
+        "rewrite-xml" to "https://github.com/openrewrite/rewrite/blob/main/rewrite-xml/src/main",
+        "rewrite-yaml" to "https://github.com/openrewrite/rewrite/blob/main/rewrite-yaml/src/main",

```

Comment:
Replace `/blob/` with `/tree/` here to skip the GitHub redirect.. blob is for files, tree is for folders. :)

Reply:
I am linking to files, though. This will be combined with stuff below to get an exact link to the file. If I used `/tree/` here, the file links won't work.

Reply:
Great work!